internal/config/config.go

```diff
@@ -328,6 +328,17 @@ func Load(cfg *Config) *viper.Viper {
 		slog.Warn("Mapping to configuration", slog.Any("error", err))
 	}
 
+	if err := Validate(cfg); err != nil {
+		reportError(err)
+		panic("Invalid configuration")
+	}
+
+	sec, err := secrets.NewAppSecrets(cfg.Application.MainSecret)
+	if err != nil {
+		panic(err)
+	}
+	cfg.Secrets = *sec
+
```

Comment:
As discussed in the other PR, this section should not be moved.

Reply:
thanks! Restoring!